src/arrayMethodSort.js

```diff
@@ -1,11 +1,32 @@
+/* eslint-disable no-param-reassign */
 'use strict';
 
 /**
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (typeof compareFunction !== 'function') {
+      compareFunction = function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        return strA < strB ? -1 : strA > strB ? 1 : 0;
+      };
+    }
```

Comment:
Move the function outside this function and use the default parameters to set this function if it's not passed https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters